src/Common/EnvironmentProxyConfigurationResolver.cpp

```diff
@@ -34,12 +35,22 @@ namespace
                 return std::getenv(PROXY_HTTPS_ENVIRONMENT_VARIABLE); // NOLINT(concurrency-mt-unsafe)
         }
     }
+
+    std::string getNoProxyHostsString()
+    {
+        const char * no_proxy = std::getenv(NO_PROXY_ENVIRONMENT_VARIABLE); // NOLINT(concurrency-mt-unsafe)
+
+        if (!no_proxy)
+        {
+            return "";
+        }
+
+        return no_proxy;
+    }
 }
 
-ProxyConfiguration EnvironmentProxyConfigurationResolver::resolve()
+ProxyConfiguration EnvironmentProxyConfigurationResolver::buildProxyConfiguration(Protocol protocol, const char * proxy_host, const std::string & no_proxy_hosts_string)
```

Comment:
Making it a member method just because of the `useTunneling` method bothers me a bit. I'll see if I can find an alternative tomorrow